### ISSUE TITLE
support customisable externalHelpersModuleNameText for --importHelpers flag 

### ISSUE THREAD

**Issue by Hotell** [Microsoft/TypeScript#12502]
**TypeScript Version:**  2.1.1 

Currently we can use `--noEmitHelpers` with `--importHelpers` to mitigate generation of bloated code in ever file which uses TS helpers ( `extends`, `decorate` ... etc ).
To make this work, currently only official [`tslib`](https://github.com/Microsoft/tslib) is supported which is [hardcoded here](https://github.com/Microsoft/TypeScript/blob/master/src/compiler/utilities.ts#L5)

if I wanna override any of these helpers I cannot use  `--importHelpers`, instead I need to import whole `tslib` and override helpers that I need before any code in app is executed, which is not very convenient, like here:

```ts
import 'tslib';

// Babel implementation of _inherits
function _inherits(subClass, superClass) {
  if (typeof superClass !== "function" && superClass !== null) {
    throw new TypeError("Super expression must either be null or a function, not " + typeof superClass);
  }
  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: {
      value: subClass,
      enumerable: false,
      writable: true,
      configurable: true
    }
  });
  if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;
}

(window as any).__extends = _inherits;
```

What I would like to have is flag in `compilerOptions` for setting node_module library name which should be used for `--importHelpers`.

Something like:

`npm install --save my-tslib`

```json
{
  "compilerOptions": {
     "module": "es2105",
     "target": "es5",
     "noEmitHelpers": true,
     "importHelpers": true,
     "importHelpersLibrary": "my-tslib"
  }
}
```

```ts
// Hello.tsx
import * as React from 'react';
class Hello extends React.Component {}
```

will compile to:
```js
// Hello.jsx
import * as tslib_1 from 'my-tslib';
import * as React from 'react';

var Hello = (function (_super) {
  tslib_1.__extends(Counter, _super);
  function Counter() {
        var _this = _super.apply(this, arguments) || this;
        return _this;
  }
  return Hello;
}(React.Component));
```

This will also help to support libraries like skate.js and use for instance Babel `extends` behaviour to make it work https://github.com/skatejs/skatejs/issues/936

**Comment by jods4** [comment]
Aurelia is a framework made of lots of independent modules and distributed in compiled form to several platforms (e.g. NPM, Bower, stand-alone). It is currently compiled with Babel but there is momentum to switch to TS.

One outstanding issue is that within this context, we don't want to take an external dependency (on `tslib`) but of course we don't want to repeat helpers all over the place either.

If we could specify the helpers module name in an option, then we could package them inside one of our core modules and have all bundles import from that. 

This way helpers are not repeated and we still have no external dependencies.

**Comment by ruizb** [comment]
I know the issue is rather old but, does anyone from the TypeScript maintainers team approve this feature?

I'd be happy to try implementing it, but as mentioned in the [CONTRIBUTING](https://github.com/microsoft/TypeScript/blob/master/CONTRIBUTING.md#contributing-features), it would be great to have an approval first 🙂 

FYI, in the project I'm currently working on (embedded in thousands of websites), we would like to provide our own implementation for some of these helpers, in order to make sure we don't use some objects/functions (such as `Object.assign`) that could be overridden by a 3rd-party script loaded in the same page as our script.
